src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2352 @@
+
+    if (count < 1) {
+        auto scope = DECLARE_THROW_SCOPE(vm);
+        JSC::throwTypeError(globalObject, scope, "createPrivateKey requires 1 arguments"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    
+    JSValue optionsArg = callFrame->uncheckedArgument(0);
+
+    if (auto* options = jsDynamicCast<JSC::JSObject*>(callFrame->argument(0))) {
+
+        JSValue keyJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "key"_s)));
+        if (keyJSValue.isUndefinedOrNull() || keyJSValue.isEmpty()) {
+            auto scope = DECLARE_THROW_SCOPE(vm);
+            JSC::throwTypeError(globalObject, scope, "key is required"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+
+        JSValue formatJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "format"_s)));
+        if (formatJSValue.isUndefinedOrNull() || formatJSValue.isEmpty()) {
+            auto scope = DECLARE_THROW_SCOPE(vm);
+            JSC::throwTypeError(globalObject, scope, "format is required"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+
+        Zig::GlobalObject* zigGlobalObject = reinterpret_cast<Zig::GlobalObject*>(globalObject);
+        auto* structure = zigGlobalObject->JSCryptoKeyStructure();
+
+        void* data;
+        size_t byteLength;
+
+        if (keyJSValue.isCell()) {
+            auto type = keyJSValue.asCell()->type();
+
+            switch (type) {
+
+            case DataViewType:
+            case Uint8ArrayType:
+            case Uint8ClampedArrayType:
+            case Uint16ArrayType:
+            case Uint32ArrayType:
+            case Int8ArrayType:
+            case Int16ArrayType:
+            case Int32ArrayType:
+            case Float32ArrayType:
+            case Float64ArrayType:
+            case BigInt64ArrayType:
+            case BigUint64ArrayType: {
+                JSC::JSArrayBufferView* view = jsCast<JSC::JSArrayBufferView*>(keyJSValue.asCell());
+
+                data = view->vector();
+                byteLength = view->length();
+                break;
+            }
+            case ArrayBufferType: {
+                auto* jsBuffer = jsCast<JSC::JSArrayBuffer*>(keyJSValue.asCell());
+                if (UNLIKELY(!jsBuffer)) {
+                    auto scope = DECLARE_THROW_SCOPE(vm);
+                    throwException(globalObject, scope, createTypeError(globalObject, "ERR_INVALID_ARG_TYPE: expected key to be Buffer or array-like object"_s));
+                    return JSValue::encode(JSC::jsUndefined());
+                }
+                auto* buffer = jsBuffer->impl();
+                data = buffer->data();
+                byteLength = buffer->byteLength();
+                break;
+            }
+            default: {
+                if (auto* keyObj = jsDynamicCast<JSC::JSObject*>(keyJSValue)) {
+                    auto format = formatJSValue.toWTFString(globalObject);
```

Comment:
needs a `RETURN_IF_EXCEPTION` check aftger this incase the `formatJSValue`.toString() call throws